pkgs/development/libraries/physics/geant4/default.nix

```diff
@@ -36,14 +36,23 @@
 }:
 
 stdenv.mkDerivation rec {
-  version = "10.5.1";
+  version = "10.6.0";
   pname = "geant4";
 
   src = fetchurl{
-    url = "http://cern.ch/geant4-data/releases/geant4.10.05.p01.tar.gz";
-    sha256 = "f4a292220500fad17e0167ce3153e96e3410ecbe96284e572dc707f63523bdff";
+    url = "https://geant4-data.web.cern.ch/geant4-data/releases/geant4.10.06.tar.gz";
+    sha256 = "169ikv2sssfbqml7bs146dj035xifxm9b12r4rzmgpvswfhca90l";
   };
 
+  # This patch fixes crash when set -u is enabled
+  patches = [
+    (fetchpatch {
+      name = "bash-variable-fix.patch";
+      url = "https://bugzilla-geant4.kek.jp/attachment.cgi?id=603&action=diff&collapsed=&headers=1&format=raw";
```

Comment:
I believe this patch is incorrect. It produces a bash code that looks like
```bash
if [ -z "${PATH+}" ] ; then
  export PATH="$geant4_envbindir"
else
  export PATH="$geant4_envbindir":${PATH}
fi
```
The if statement is never false and that breaks the environment:
```
% [ -z "${PATH+}" ] && echo q
q
% [ -z "${NOTATHING+}" ] && echo q
q
```
You probably want `${PATH-}`, e.g. see https://wiki.bash-hackers.org/syntax/pe#use_a_default_value

Reply:
oof.